tests/wrappers/test_fastapi_wrapper.py

```diff
@@ -130,24 +254,50 @@ async def test_fastapi_multiple_requests(_, trace_transport, fastapi_app):
         )
 
 
-@pytest.mark.asyncio
-@asynctest.patch('epsagon.trace.trace_factory.use_async_tracer')
-async def test_fastapi_multiple_threads_route(_, trace_transport, fastapi_app):
+
+async def _send_async_request(app, path):
+    async with AsyncClient(app=app, base_url="http://test") as ac:
+        return await ac.get(path)
+
+@pytest.mark.parametrize(
+    "fastapi_app",
+    [
+        pytest.lazy_fixture("sync_fastapi_app"),
+        pytest.lazy_fixture("async_fastapi_app"),
+    ],
+)
+def test_fastapi_multiple_threads_route(trace_transport, fastapi_app):
     """
     Tests request to a route, which invokes multiple threads.
     Validating no `zombie` traces exist (fromn the callback invoked threads)
     """
-    async with AsyncClient(app=fastapi_app, base_url="http://test") as ac:
-        response = await ac.get(f"{MULTIPLE_THREADS_ROUTE}?x=testval")
+    # used to reset trace factory between fastapi apps
+    trace_factory.use_single_trace = True
```

Comment:
maybe wrap with a `set_async_fast_api` / `set_sync_fast_api` functions? it will make the test more readable - now we're setting this mode or another

Reply:
fixed with a setup function